src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioTaskSchedulerFactory.cs

```diff
@@ -30,6 +30,11 @@ public VisualStudioTaskSchedulerFactory(IThreadingContext threadingContext, IAsy
 
         public override IWorkspaceTaskScheduler CreateEventingTaskQueue()
         {
+            if (!_threadingContext.HasMainThread)
```

Comment:
"HasMainThread" means "we have a concept of a main thread, but we're not saying if we're on it or not?"?

Reply:
Exactly correct

Reply:
This change is now removed from this pull request.